client/layout/guided-tours/index.js

```diff
@@ -59,7 +60,22 @@ class GuidedTours extends Component {
 
 	next() {
 		const nextStepName = this.props.tourState.stepConfig.next;
-		this.props.nextGuidedTourStep( { stepName: nextStepName } );
+		const predicate = () => {
+			console.log( 'predicate' );
+			const { stepConfig } = this.props.tourState;
+			if ( stepConfig && stepConfig.target === 'my-sites' ) {
```

Comment:
Shouldn't we look at the *next* step to find if it has a target, and if so try to look it up?